app/middleware/auth.py

```diff
@@ -0,0 +1,55 @@
+from fastapi import Request
+from starlette.middleware.base import BaseHTTPMiddleware
+from app.core.security import get_current_user_from_auth
+from app.db.database import get_db
+import logging
+from typing import Optional, Dict, Any
+
+logger = logging.getLogger(__name__)
+
+class AuthMiddleware(BaseHTTPMiddleware):
+    async def dispatch(self, request: Request, call_next):
+        # Skip auth for certain paths
+        if request.url.path in ["/health", "/docs", "/openapi.json", "/metrics"]:
```

Comment:
super nit picky but could maybe benefit from being extracted to `const.PUBLIC_ENDPOINTS` or similar and then using that here and [here](https://github.com/amazeeio/amazee.ai/pull/43/files#diff-0fdfa731bba127e9566ed41eba69ab3eaa6f3439403d1b457a92d92e7dcf8473R23), just imagining it'd be easier to manage centrally whenever the list changes...